src/drivers/indexeddb.js

```diff
@@ -318,19 +320,22 @@ function _tryReconnect(dbInfo) {
             forage._dbInfo.db = null;
         }
     }
+    dbInfo.db = null;
 
     return _getOriginalConnection(dbInfo)
         .then(function(db) {
-            for (var j = 0; j < forages.length; j++) {
-                forages[j]._dbInfo.db = db;
-            }
             dbInfo.db = db;
-        })
-        .then(function() {
             if (_isUpgradeNeeded(dbInfo)) {
                 // Reopen the database for upgrading.
                 return _getUpgradedConnection(dbInfo);
             }
+            return db;
+        })
+        .then(function(db) {
+            dbInfo.db = dbContext.db = db;
+            for (var j = 0; j < forages.length; j++) {
```

Comment:
Works be really happy to do such a thing.
What do you have in mind?
Last time I checked `.forEach` wasn't getting polyfilled in this repo and `for of` had an other issue (will try again later).

Reply:
Perhaps in v2.0 we could have one build with polyfills (not just Promise) and one without. We should probably also default to the non polyfilled version. I've read about people being concerned of the main dist file size without knowing that we have a nopolyfills variant.

Reply:
@tofumatt Using babel es2015-loose [`for of` & `.forEach()`](https://babeljs.io/repl/#?babili=false&browsers=IE%208&build=&builtIns=usage&code_lz=PTAEEsFsAcHsCcAuoBEAjAhmgpgGwLRy4CeAZuLrigNwBQtpArgHYDGi4szopAFOAEpQAb1qhQrLgGdYubADpcsAOb8BdAL71JzKcgBeoALygA2gHIM5gDTm0N863MBdOqQSheOvaAAeoWFJQfSFRAEg-X3VaLQYPXjlkcGNQAAZqCFAAHmDFbGZlRAALDPAAajLQ2jCpbEQAFShsWEZEXl4hIwA-HjVo2P15SAxodqjjHsiBaKA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=true&lineWrap=true&presets=es2015-loose&prettier=false&targets=&version=6.26.0&envVersion=1.6.2) works on IE9+,  as long as you only use arrays, otherwise the reference to `Symbol` in the transpiled code will error when it gets evaluated.
My concern (which holded me back all this time) include:
* is the extra file size increase of the transpiled `for...of` really worth it?
* should we start using untranspiled es5 Array.* methods?

PS: Typescript (because of the type inference) [emits far less & cleaner code for `for...of`](https://www.typescriptlang.org/play/index.html). Should we consider it at some point? Especially editing this driver would feel safer.